sherpa-onnx/csrc/funasr-nano-tokenizer.cc

```diff
@@ -0,0 +1,1366 @@
+            j += nx;
+          }
+          while (j < text.size()) {
+          size_t t = j;
+          uint32_t cx = 0;
+          size_t nx = 0;
+          if (!Utf8Next(text, &t, &cx, &nx)) break;
+          if (!IsNewline(cx)) break;
+          j += nx;
+        }
+        out.push_back(text.substr(start, j - start));
+        i = j;
+        continue;
+      }
+    }
+
+    {
+      if (IsWhitespace(cp)) {
+        size_t start = i;
+        size_t j = i;
+
+        bool saw_newline = false;
+        while (j < text.size()) {
+          size_t t = j;
+          uint32_t cx = 0;
+          size_t nx = 0;
+          if (!Utf8Next(text, &t, &cx, &nx)) break;
+          if (IsNewline(cx)) {
+            saw_newline = true;
+            break;
+          }
+          if (!IsWhitespace(cx)) break;
+          j += nx;
+        }
+
+        if (saw_newline) {
+          while (j < text.size()) {
+            size_t t = j;
+            uint32_t cx = 0;
+            size_t nx = 0;
+            if (!Utf8Next(text, &t, &cx, &nx)) break;
+            if (!IsNewline(cx)) break;
+            j += nx;
+          }
+          out.push_back(text.substr(start, j - start));
+          i = j;
+          continue;
+        }
+      }
+    }
+
+    if (IsWhitespace(cp)) {
+      bool only_ws_to_end = true;
+      size_t j = i;
+      while (j < text.size()) {
+        size_t t = j;
+        uint32_t cx = 0;
+        size_t nx = 0;
+        if (!Utf8Next(text, &t, &cx, &nx)) break;
+        if (!IsWhitespace(cx)) {
+          only_ws_to_end = false;
+          break;
+        }
+        j += nx;
+      }
+      if (only_ws_to_end) {
+        out.push_back(text.substr(i));
+        break;
+      }
+    }
+
+    if (IsWhitespace(cp)) {
+      size_t start = i;
+      size_t j = i;
+      while (j < text.size()) {
+        size_t t = j;
+        uint32_t cx = 0;
+        size_t nx = 0;
+        if (!Utf8Next(text, &t, &cx, &nx)) break;
+        if (!IsWhitespace(cx)) break;
+        j += nx;
+      }
+      out.push_back(text.substr(start, j - start));
+      i = j;
+      continue;
+    }
+
+    out.push_back(text.substr(i, n));
+    i += n;
+  }
+
+  return out;
+}
+
+static std::vector<std::string> SplitUtf8ToChars(const std::string &s) {
+  std::vector<std::string> out;
+  out.reserve(s.size());
+  size_t i = 0;
+  while (i < s.size()) {
+    size_t t = i;
+    uint32_t cp = 0;
+    size_t n = 0;
+    if (!Utf8Next(s, &t, &cp, &n) || n == 0) {
+      out.push_back(s.substr(i, 1));
+      i += 1;
+      continue;
+    }
+    out.push_back(s.substr(i, n));
+    i += n;
+  }
+  return out;
+}
+
+static inline std::string MakeMergeKey(const std::string &a,
+                                       const std::string &b) {
+  std::string k = a;
+  k.push_back('\t');
+  k.append(b);
+  return k;
+}
+
+}  // namespace
+
+// Parse tokenizer.json added_tokens: extract objects with {id, content, ...}
+bool ParseAddedTokensFromTokenizerJson(
```

Comment:
```suggestion
static bool ParseAddedTokensFromTokenizerJson(
```

or move it into the above anonymous namespace.